test/unit/network/ConnectionRegistryTest.js

```diff
@@ -0,0 +1,145 @@
+                connectionRegistry.setConnection(UuidUtil.generate(), {});
+                connectionRegistry.setConnection(UuidUtil.generate(), {});
+
+                const connection = connectionRegistry.getRandomConnection();
+                const otherConnection = connectionRegistry.getRandomConnection();
+                const anotherConnection = connectionRegistry.getRandomConnection();
+
+                expect(connection).to.be.equal(firstConnection);
+                expect(otherConnection).to.be.equal(firstConnection);
+                expect(anotherConnection).to.be.equal(firstConnection);
+
+                expect(loadBalancerStub.next.called).to.be.false;
+            }
+        );
+    });
+
+    describe('checkIfInvocationAllowed', function () {
+        it('should return null when ConnectionState=INITIALIZED_ON_CLUSTER and there are some active connections',
+            function () {
+                const connectionRegistry = new ConnectionRegistryImpl(
+                    new ConnectionStrategyConfigImpl(),
+                    false,
+                    new RoundRobinLB()
+                );
+
+                connectionRegistry.setConnectionState(connectionState.INITIALIZED_ON_CLUSTER);
+                connectionRegistry.setConnection(UuidUtil.generate(), new sinon.createStubInstance(Connection));
+
+                expect(connectionRegistry.checkIfInvocationAllowed()).to.be.equal(null);
+            }
+        );
+
+        it('should return ClientOfflineError when ConnectionState=INITIAL and asyncStart=true', function () {
+            const connectionStrategyConfig = new ConnectionStrategyConfigImpl();
+            connectionStrategyConfig.asyncStart = true;
+
+            const connectionRegistry = new ConnectionRegistryImpl(
+                connectionStrategyConfig,
+                false,
+                new RoundRobinLB()
+            );
+
+            connectionRegistry.setConnectionState(connectionState.INITIAL);
+
+            expect(connectionRegistry.checkIfInvocationAllowed()).to.be.instanceof(ClientOfflineError);
+        });
+
+        it('should return IOError when ConnectionState=INITIAL and asyncStart=false', function () {
+            const connectionStrategyConfig = new ConnectionStrategyConfigImpl();
+            connectionStrategyConfig.asyncStart = false;
+
+            const connectionRegistry = new ConnectionRegistryImpl(
+                connectionStrategyConfig,
+                false,
+                new RoundRobinLB()
+            );
+
+            connectionRegistry.setConnectionState(connectionState.INITIAL);
+
+            expect(connectionRegistry.checkIfInvocationAllowed()).to.be.instanceof(IOError);
+        });
+
+        it('should return ClientOfflineError when reconnectMode=ReconnectMode.ASYNC mode, '
+         + 'ConnectionState=INITIALIZED_ON_CLUSTER and there are no connections',
+        function () {
```

Comment:
found this strange at first, but this is the correct indentation according to eslint when first argument is multiline see https://github.com/eslint/eslint/issues/9706#issuecomment-351290798